File: 백준/Gold/1068. 트리/트리.cc
#include<bits/stdc++.h>
using namespace std;
int n,remove_node,node,root;
vector<int> adj[54];
int dfs(int here)
{
	int ret=0;
	int child=0;
	for(int there : adj[here])
	{
		if(there==remove_node) continue;
		ret+=dfs(there);
		child++;
	}
	if(child==0) return 1;
	return ret;
}
int main()
{
	cin>>n;
	for(int i=0;i<n;i++)
	{
		cin>>node;
		if(node==-1) root=i;
		else adj[node].push_back(i);
	}
	cin>>remove_node;
	if(remove_node==root) cout<<0;
	else cout<<dfs(root);
}

File: 백준/Gold/1068. 트리/트리.py
n=int(input().strip())
nodes=list(map(int,input().split()))
remove_node=int(input().strip())

tree=[[] for _ in range(n)]
root=-1
for i in range(n):
    if nodes[i]==-1:
        root=i
    else:
        tree[nodes[i]].append(i) # 각 노드의 자식 노드 번호 기록
        
def dfs(here):
    ret=child=0
    for there in tree[here]:
        if there==remove_node:
            continue
        ret+=dfs(there)
        child+=1

    if child==0:
        ret=1

    return ret

if remove_node==root:
    print(0)
else:
    print(dfs(root))